sdk/storage/azure-storage-file-share/azure/storage/fileshare/_lease.py

```diff
@@ -137,7 +201,15 @@ def change(self, proposed_lease_id, **kwargs):
                 proposed_lease_id=proposed_lease_id,
                 timeout=kwargs.pop('timeout', None),
                 cls=return_response_headers,
+                **kwargs) if isinstance(self._client, FileOperations) \
+                else self._client.change_lease(
+                lease_id=self.id,
+                proposed_lease_id=proposed_lease_id,
+                timeout=kwargs.pop('timeout', None),
+                sharesnapshot=self.snapshot,
+                cls=return_response_headers,
                 **kwargs)
```

Comment:
you don't have to check if it's a FileOperations instance, since when you call acquire_lease on the given self._client, the corresponding acquire_lease will be auto detected and called no matter it's in FileOperations or ShareOperations

Reply:
Yes, I'm aware of that however the reason I'm checking if its a FileOperation or a ShareOperation is because the spec stated that only Shares can pass snapshots and files cannot

Reply:
if you pass sharesnapshot=self.snapshot from ShareClient, then lease_client.acuqire_lease will have that snapshot info in kwargs.

Reply:
you are absolutely right, I completely forgot about this! thank you :)

Reply:
Another thing, we fileshare doesn't support write, I'm still thinking if it's necessary to acquire lease on a snapshot

Reply:
BTW currently, the code doesn't work, since ShareClient isn't passing sharesnapshot=self.snapshot when calling lease_client.acquire_lease() internally, while it doesn't matter, as long as we detect it in LeaseClient.